### PR TITLE
4613 doc sign validation

### PR DESCRIPTION
**Pull request by matthopson**
This adds some validation when creating a docket entry with a document eventCode that requires a signature to prevent the create docket entry form from being submitted, as well as addresses some UX feedback regarding some flow and UI.

### CODE REVIEW

File: web-client/src/presenter/actions/CourtIssuedDocketEntry/validateCourtIssuedDocketEntryAction.js [hunk 1]
```diff
@@ -14,15 +14,33 @@ export const validateCourtIssuedDocketEntryAction = ({
...
     .getUseCases()
     .validateCourtIssuedDocketEntryInteractor({
       applicationContext,
       entryMetadata,
     });
 
+  // Additional validation to determine if the signature required warning should be displayed
+  if (EVENT_CODES_REQUIRING_SIGNATURE.includes(entryMetadata.eventCode)) {
+    const document = caseDetail.documents.find(
+      document => document.documentId === documentId,
```
Comment (rachaelparris) [inline web-client/src/presenter/actions/CourtIssuedDocketEntry/validateCourtIssuedDocketEntryAction.js]: You'll get a sonarcloud error about this, using the variable `document` again. 😄

File: web-client/src/presenter/computeds/messageDocumentHelper.js [hunk 2]
```diff
@@ -31,19 +36,26 @@ export const messageDocumentHelper = (get, applicationContext) => {
   const showEditButtonForRole = isInternalUser;
   const showApplyEditSignatureButtonForRole = isInternalUser;
 
-  const showAddDocketEntryButtonForDocument = !isDocumentOnDocketRecord;
+  const showAddDocketEntryButtonForDocument =
+    !isDocumentOnDocketRecord &&
+    ((documentRequiresSignature && documentIsSigned) ||
+      !documentRequiresSignature);
```
Comment (rachaelparris) [inline web-client/src/presenter/computeds/messageDocumentHelper.js]: Nit: can be simplified to `(documentIsSigned || !documentRequiresSignature)`

File: web-client/src/views/DocketRecord/DraftDocumentViewerDocument.jsx [hunk 3]
```diff
@@ -45,14 +45,22 @@ export const DraftDocumentViewerDocument = connect(
...
+                </div>
+              )}
             </div>
 
             <div className="message-document-actions">
               {draftDocumentViewerHelper.showEditButtonNotSigned && (
                 <Button
                   link
-                  href={`/case-detail/${caseDetail.docketNumber}/edit-order/${viewerDraftDocumentToDisplay.documentId}`}
+                  href={viewerDraftDocumentToDisplay.editUrl}
```
Comment (rachaelparris) [inline web-client/src/views/DocketRecord/DraftDocumentViewerDocument.jsx:63]: Does this URL change based on any logic? If not, I think it's more clear to have it hardcoded here. I tried figuring out where `editUrl` was coming from and it really wasn't clear.

Comment (matthopson) [reply web-client/src/views/DocketRecord/DraftDocumentViewerDocument.jsx:63]: We don't want to take the user to the `edit-order` route for uploaded PDFs.

**Review by rachaelparris — APPROVED** [review]

**Review by kkoskelin — APPROVED** [review]

**Pull request merged**